LibraryApp/src/components/books-list.js

```diff
@@ -1,41 +1,133 @@
-import React, {useState} from 'react';
+import React, {useState, useEffect} from 'react';
 import {
   StyleSheet,
   View,
   Text,
   TextInput,
   SafeAreaView,
   Pressable,
+  Platform,
+  ActivityIndicator,
 } from 'react-native';
 import {connect} from 'react-redux';
-import {setSelectedBook} from 'store/actions/library';
+import Geolocation from '@react-native-community/geolocation';
 import {Colors, Global, Spacing, Typography} from 'stylesheets';
+import PermissionService from 'services/permissions';
+import {setSelectedBook, setCurrentLocation} from 'store/actions/library';
 import {libraryForm, showBookDetails, scanQRCode} from 'constants/navigators';
 import {FilterListBySearch} from 'helpers/library';
-import {search, qrCode} from 'constants/icons';
+import {search, qrCode, refresh} from 'constants/icons';
 import List from 'components/shared/list';
+import {LocationStatus} from 'constants/location-status';
 import {useHardwareBack} from 'components/custom/hardware-back';
 import {CustomImage} from 'components/shared/common';
 
 const isIOSPlatform = Platform.OS == 'ios';
 const loaderTimeout = 2000;
+const defaultObj = {};
 
 const {
   flex1,
   rowFlex,
+  spaceBetween,
   borderRadius20,
   horizontalCenter,
   verticalCenter,
 } = Global;
-const {comicFont, gochiFont, iconFont, fs18, fs20, fs25, bold} = Typography;
+const {
+  comicFont,
+  gochiFont,
+  iconFont,
+  fs16,
+  fs18,
+  fs20,
+  fs25,
+  bold,
+} = Typography;
 const {p10, py20, px10, px15, px25, p15, mtAuto, mb15} = Spacing;
-const {blue, darkGrey, purple, lightGreen, maroon, white} = Colors;
+const {blue, darkGrey, purple, lightGreen, maroon, skyBlue, white} = Colors;
+const {loading, loaded, denied} = LocationStatus;
+
+const Options = {
+  currentLocation: {
+    enableHighAccuracy: false,
+    timeout: 30000,
+    maximumAge: 1000,
+  },
+  subscribeLocation: {
+    enableHighAccuracy: false,
+    maximumAge: 1000,
+    distanceFilter: 1,
+  },
+};
 
 const getItemKey = (item) => `book-${item.id}`;
 
 const BooksList = (props) => {
   const [isRefreshing, setIsRefreshing] = useState(false);
   const [searchedText, setSearchedText] = useState('');
+  const [locationStatus, setLocationStatus] = useState('');
+  let watchID;
+
+  useEffect(() => {
```

Comment:
You are tracking the location on BookListing but showing the distance in BookShow, how will that work? Shouldn't we track the location on book show only?

Reply:
Created separate current-location component and called on showBookDetails component instead of BookListing.